pkg/node/manager/node_config_notifier.go

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package manager
+
+import (
+	"errors"
+
+	datapath "github.com/cilium/cilium/pkg/datapath/types"
+)
+
+type NodeConfigNotifier struct {
+	handlers []datapath.NodeConfigChangeHandler
+}
+
+func newNodeConfigNotifier() *NodeConfigNotifier {
+	return &NodeConfigNotifier{
+		handlers: make([]datapath.NodeConfigChangeHandler, 0),
+	}
+}
+
+func (n *NodeConfigNotifier) Subscribe(handler datapath.NodeConfigChangeHandler) {
+	n.handlers = append(n.handlers, handler)
+}
+
+func (n *NodeConfigNotifier) Unsubscribe(handler datapath.NodeConfigChangeHandler) {
+	for i, h := range n.handlers {
```

Comment:
nit: We could use `slices.DeleteFunc` here. But overall the unsubscribe logic is not even used. Why implement it?

Reply:
I can't even recall why I added this, mostly because my brain went into "implement subscriber patten" mode. Will remove :+1: